### PR TITLE
mentorship tasks

### PR DESCRIPTION
**Pull request by kengthong**
# Feature

Use Case: D.1.10 View Mentorship Tasks
Feature ID : _Eg; 1_
Feature: _Eg; Sensei/Student Account Access_
Figma link for wireframe:

# Changelog:

# Merge/Review Order

These other PRs should be merged or reviewed first

This PR blocks these other PRs

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

### CODE REVIEW

**Review by rawrrencee — DISMISSED** [review]
@natalieloke for your review~ I think code is ok! but need your help to test the business logic part of this to see if any changes will b required. thanks!

File: src/components/Mentorship/Subscription/TaskBucket.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            onKeyDown={() => showTasks(tb)}
+            className={`clickable bucket-items d-flex justify-content-between ${
+              tb.taskBucketId === activeTaskBucket.bucket?.taskBucketId ? 'bucket-item-active' : ''
+            } col-12`}
+            style={{ padding: '4px 8px' }}
+            key={tb.taskBucketId}
+          >
+            {tb.title}
+            <Popconfirm
+              title="Do you wish to delete admin account?"
```
Comment (natalieloke) [inline src/components/Mentorship/Subscription/TaskBucket.js]: Need to change pasta-ed text

File: src/components/Mentorship/ListingDetails/MentorshipProfileHeader.js [hunk 2]
```diff
@@ -42,29 +42,31 @@ const MentorshipProfileHeader = () => {
...
+            type="primary"
+            size="large"
+            shape="round"
+            onClick={onAdd}
+            icon={<CheckSquareOutlined />}
+          >
+            Apply for Mentorship
+          </Button>
+        </div>
+      ) : null}
```
Comment (natalieloke) [inline src/components/Mentorship/ListingDetails/MentorshipProfileHeader.js]: If the false condition returns null will it just be easier to put` !isSubscription && <Button....>`

Also helps w readability!

File: src/components/Mentorship/Subscription/TaskBucket.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  addTaskBucket,
+}) => {
+  const [input, setInput] = useState('')
+  return (
+    <div
+      role="menu"
+      className="col-12"
+      style={{ paddingRight: '20px', borderRight: '1px solid #e8e8e8' }}
+    >
+      {taskBuckets.map((tb, i) => {
```
Comment (natalieloke) [inline src/components/Mentorship/Subscription/TaskBucket.js]: Can we use lodash's map? JavaScript map has some issues when the object is empty/null/undefined (I forgot which lol)

File: src/components/Mentorship/Subscription/TaskRow.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    </Menu>
+  )
+
+  return (
+    <div className="d-flex align-items-center justify-content-start" style={{ width: '100%' }}>
+      <div className="col-12 col-md-7 d-flex justify-content-center align-items-center">
+        <div className="col-12 col-md-4 d-flex ">
+          <Dropdown overlay={menu} trigger={['click']} placement="topRight" className="clickable">
+            <div style={{ width: '100%' }} className="d-flex align-items-center">
+              <Tag
```
Comment (natalieloke) [inline src/components/Mentorship/Subscription/TaskRow.js]: U can check out StatusTag component which show diff colours!

**Review by natalieloke — CHANGES_REQUESTED** [review]
Thanks KT 🙏🏼 Left some comments! Biz logic wise I think it seems ok from the code, a video would help! Else we can always build up on the biz logic on subsequent PRs :) 

cc @rawrrencee

**Review by natalieloke — APPROVED** [review]

File: src/components/Mentorship/ListingDetails/MentorshipProfileHeader.js [hunk 5]
```diff
@@ -32,39 +28,33 @@ const MentorshipProfileHeader = () => {
   return (
     <div className="row justify-content-between ">
```
Comment (rawrrencee) [inline src/components/Mentorship/ListingDetails/MentorshipProfileHeader.js:29]: extra space, if wanna modify can, if not its ok too

File: src/components/Mentorship/Subscription/TaskRow.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      <Menu.Item key="COMPLETED">COMPLETED</Menu.Item>
+      <Menu.Item key="ONGOING">ONGOING</Menu.Item>
+      <Menu.Item key="NOT_STARTED">NOT STARTED</Menu.Item>
+    </Menu>
+  )
+
+  return (
+    <div className="d-flex align-items-center justify-content-start" style={{ width: '100%' }}>
+      <div className="col-12 col-md-7 d-flex justify-content-center align-items-center">
+        <div className="col-12 col-md-4 d-flex ">
```
Comment (rawrrencee) [inline src/components/Mentorship/Subscription/TaskRow.js:74]: extra space here too, not necessary to fix

**Review by rawrrencee — APPROVED** [review]
looks good to me! there's some small stuff like equivalent classNames for some style={{ }} stuff, but since its equivalent, i think can leave it. there's also some div classNames="... " extra spacing behind the ..., I have left comments on them. but don't change also nvm~

**Comment by natalieloke** [comment]
> looks good to me! there's some small stuff like equivalent classNames for some style={{ }} stuff, but since its equivalent, i think can leave it. there's also some div classNames="... " extra spacing behind the ..., I have left comments on them. but don't change also nvm~

ok i'll change those along the way in subsequent PRs thanks! @rawrrencee

**Pull request merged**